sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -148,6 +149,108 @@ public TransportResult send(final SentryEvent event) throws IOException {
     }
   }
 
+  @Override
+  public boolean isRetryAfter(String type) {
+    // TODO: We need a mechanism (ConcurrentDictionary?) That will keep RetryAfter per envelope item type
+    // The Timer we use now could be used to drop keys from the hashSet:
+
+    // pseudo: return retryAfterHashSet.contains(type);
+    return false;
+  }
+
+  @Override
+  public TransportResult send(SentryEnvelope envelope) throws IOException {
+
+    // TODO: This is a huge copy paste to test things out.
+
+    final HttpURLConnection connection = open(proxy);
+    connectionConfigurator.configure(connection);
+
+    connection.setRequestMethod("POST");
+    connection.setDoOutput(true);
+    // TODO: Should do Content-Encoding=gzip
+    // TODO: Just wrap the Stream with GzipStream?
+    connection.setRequestProperty("Content-Encoding", "UTF-8");
+    connection.setRequestProperty("Content-Type", "application/x-sentry-envelope");
+    connection.setRequestProperty("Accept", "application/json");
+
+    // https://stackoverflow.com/questions/52726909/java-io-ioexception-unexpected-end-of-stream-on-connection/53089882
+    connection.setRequestProperty("Connection", "close");
+
+    connection.setConnectTimeout(connectionTimeout);
+    connection.setReadTimeout(readTimeout);
+
+    if (bypassSecurity && connection instanceof HttpsURLConnection) {
+      ((HttpsURLConnection) connection).setHostnameVerifier((__, ___) -> true);
+    }
+
+    connection.connect();
+
+    try (final OutputStream outputStream = connection.getOutputStream()) {
+//      DataOutputStream dataStream = new DataOutputStream(conn.getOutputStream());
+      serializer.serialize(envelope, outputStream);
+
+      // need to also close the input stream of the connection
+      connection.getInputStream().close();
+      options.getLogger().log(DEBUG, "Envelope sent %s successfully.", envelope.getHeader().getEventId());
+      return TransportResult.success();
+    } catch (IOException e) {
+      long retryAfterMs = HTTP_RETRY_AFTER_DEFAULT_DELAY_MS;
+      final String retryAfterHeader = connection.getHeaderField("Retry-After");
+      if (retryAfterHeader != null) {
+        // TODO: Protocol undefined yet but ...
+        String rateLimitDetailString = connection.getHeaderField("X-Sentry-Rate-Limit");
```

Comment:
wait `Retry-After` and `X-Sentry-Rate-Limit`, I need to update myself about `X-Sentry-Rate-Limit`

Reply:
It's in `Rate Limiting Updates` internal doc